src/components/configMaps/ConfigMap.tsx

```diff
@@ -233,35 +235,75 @@ export function ConfigMapForm({ id, appId, name = "", external, data = null, typ
+                await deleteConfig(id, appId, name)
                 toast.success('Successfully deleted')
                 updateForm(listIndex, null)
             }
             else {
-                const { result } = await deleteEnvironmentConfig(id, appId, envId, name);
+                await deleteEnvironmentConfig(id, appId, envId, name);
                 toast.success('Successfully deleted')
                 updateForm(true)
             }
-
         }
         catch (err) {
             showError(err)
         }
     }
 
+    function handleFilePermission(e): void {
+        let permissionValue = e.target.value;
+        setFilePermissionValue({ value: permissionValue, error: "" });
+    }
+
     async function handleSubmit(e) {
+        const configmapNameRegex = new RegExp(PATTERNS.CONFIGMAP_AND_SECRET_NAME);
         if (!configName.value) {
-            setName({ value: "", error: 'Field is manadatory' })
+            setName({ value: "", error: 'This is a required field' })
             return
         }
-        if (!/^[-.a-zA-Z0-9]+$/.test(configName.value)) {
-            setName({ value: configName.value, error: 'Name must be of format /^[-.a-zA-Z0-9]+$/' })
+        if (configName.value.length > 253) {
+            setName({ value: configName.value, error: 'More than 253 characters are not allowed' })
+            return
+        }
+        if (!configmapNameRegex.test(configName.value)) {
+            setName({ value: configName.value, error: `Name must start and end with an alphanumeric character. It can contain only lowercase alphanumeric characters, '-' or '.'` })
             return
         }
         if (selectedTab === 'Data Volume' && !volumeMountPath.value) {
-            setVolumeMountPath({ value: volumeMountPath.value, error: 'Field is manadatory' })
+            setVolumeMountPath({ value: volumeMountPath.value, error: 'This is a required field' })
             return
         }
+        if (selectedTab === 'Data Volume' && isFilePermissionChecked) {
+            if (!filePermissionValue.value) {
+                setFilePermissionValue({ value: filePermissionValue.value, error: 'This is a required field' });
+                return;
+            }
+            else if (filePermissionValue.value.length > 4) {
+                setFilePermissionValue({ value: filePermissionValue.value, error: 'More than 4 characters are not allowed' });
+                return;
+            }
+            else if (filePermissionValue.value.length === 4) {
+                if (!filePermissionValue.value.startsWith('0')) {
+                    setFilePermissionValue({ value: filePermissionValue.value, error: '4 characters are allowed in octal format only, first character should be 0' });
+                    return;
+                }
+            }
+            else if (filePermissionValue.value.length === 3) {
+                if (filePermissionValue.value.startsWith('0')) {
```

Comment:
remove this failure condition, 000 is valid

Reply:
removed